File: src/test/java/de/poeschl/bukkit/logcleaner/helper/FileHelperTest.java
package de.poeschl.bukkit.logcleaner.helper;

import org.apache.commons.lang.time.DateUtils;
import org.hamcrest.collection.IsIterableContainingInOrder;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mockito;

import java.io.File;
import java.io.FilenameFilter;
import java.util.Calendar;
import java.util.logging.Logger;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertThat;
import static org.mockito.Matchers.any;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class FileHelperTest {

    private FileHelper testFileHelper;

    @Before
    public void setUp() {
        Logger voidLogger = Mockito.mock(Logger.class);
        testFileHelper = new FileHelper(voidLogger);
    }

    @Test
    public void getLogArchives() throws Exception {
        File mockFolder = Mockito.mock(File.class);
        File[] mockFileList = new File[]{new File("a.log.gz"), new File("b.log.gz")};
        when(mockFolder.listFiles(any(FilenameFilter.class))).thenReturn(mockFileList);

        assertThat(testFileHelper.getLogArchives(mockFolder),
                IsIterableContainingInOrder.contains(mockFileList));
    }

    @Test
    public void getLogArchiveDate() throws Exception {
        File mockArchive = Mockito.mock(File.class);
        when(mockArchive.getName()).thenReturn("1234-12-12-9.log.gz");

        Calendar expected = Calendar.getInstance();
        expected.set(1234, Calendar.DECEMBER, 12);


        assertEquals(DateUtils.truncate(expected.getTime(), Calendar.DAY_OF_MONTH),
                DateUtils.truncate(testFileHelper.getLogArchiveDate(mockArchive), Calendar.DAY_OF_MONTH));
    }

    @Test
    public void delete() throws Exception {
        File mockFile = Mockito.mock(File.class);

        testFileHelper.delete(mockFile);
        verify(mockFile).delete();
    }

}

File: src/test/kotlin/de/poeschl/bukkit/logcleaner/helper/FileHelperTest.kt
package de.poeschl.bukkit.logcleaner.helper

import com.nhaarman.mockito_kotlin.mock
import com.nhaarman.mockito_kotlin.whenever
import de.poeschl.bukkit.logcleaner.createDate
import org.assertj.core.api.Assertions
import org.assertj.core.util.DateUtil
import org.junit.Test
import org.mockito.ArgumentMatchers
import org.mockito.Mockito.verify
import java.io.File
import java.util.*
import java.util.logging.Logger


class FileHelperTest {

    @Test
    fun getLogArchives() {
        val mockFolder: File = mock()
        val mockFileList = arrayOf(File("a.log.gz"), File("b.log.gz"))
        val voidLogger: Logger = mock()
        val testFileHelper = FileHelperImpl(voidLogger)
        whenever(mockFolder.listFiles(ArgumentMatchers.any(java.io.FilenameFilter::class.java))).thenReturn(mockFileList)

        val resultFiles = testFileHelper.getLogArchives(mockFolder)

        Assertions.assertThat(resultFiles).containsAll(mockFileList.asList())
    }

    @Test
    fun getLogArchiveDate() {
        val mockArchive: File = mock()
        val expectedDay = DateUtil.dayOfMonthOf(Calendar.getInstance().createDate(1234, Calendar.DECEMBER, 12))
        val voidLogger: Logger = mock()
        val testFileHelper = FileHelperImpl(voidLogger)
        whenever(mockArchive.name).thenReturn("1234-12-12-9.log.gz")

        val resultDay = DateUtil.dayOfMonthOf(testFileHelper.getLogArchiveDate(mockArchive))

        Assertions.assertThat(resultDay).isEqualTo(expectedDay)
    }

    @Test
    fun delete() {
        val mockFile: File = mock()
        val voidLogger: Logger = mock()
        val testFileHelper = FileHelperImpl(voidLogger)

        testFileHelper.delete(mockFile)

        verify(mockFile).delete()
    }

}